test/testother.cpp

```diff
@@ -13093,6 +13095,19 @@ class TestOther : public TestFixture {
             "[test.cpp:2:13] -> [test.cpp:3:11]: (style) The comparison 'i > +1' is always false. [knownConditionTrueFalse]\n",
             errout_str());
     }
+
+    void ternarySameValuePlatformDependent() // #13773
+    {
+        check("typedef float _Complex complex_f32;\n"
+              "typedef struct {\n"
+              "    uint16_t real;\n"
+              "    uint16_t imag;\n"
+              "} packed_complex;\n"
+              "void foo(void) {\n"
+              "    b = a ? sizeof(packed_complex) : sizeof(complex_f32);\n"
```

Comment:
please add some testcase with duplicate `sizeof` expressions also

Reply:
hmm.. the testcase is more complex than necessary. I have the feeling we can just write something like:
```
void foo(void) {
    b = a ? sizeof(uint32_t) : sizeof(unsigned int);
}
```